cmd/main.go

```diff
@@ -72,10 +70,31 @@ func main() {
 	shutdown := make(chan os.Signal, 1)
 	signal.Notify(shutdown, syscall.SIGINT, syscall.SIGTERM)
 
-	if err := run(log, cfg, shutdown); err != nil {
-		log.Error("startup", "msg", err)
+	// serve as default to prevent breaking change
+	action := "serve"
+	if len(os.Args) > 1 {
+		action = os.Args[1]
+	}
+
+	switch action {
+	case "serve":
+		if err := httpServer(log, cfg, shutdown); err != nil {
+			log.Error("startup", "msg", err)
+			os.Exit(1)
+		}
+	case "validate":
+		if err := validate(log, cfg, shutdown); err != nil {
+			log.Error("validate", "msg", err)
+			os.Exit(1)
+		}
+	case "version":
+		log.Info("GraphQL Protect", "version", build, "go_version", runtime.Version(), "short_hash", shortHash)
+		os.Exit(0)
+	default:
```

Comment:
Why not make serve the default path as well? That prevents the manual `action`

Reply:
thought about that, but that'd also make `asdf` do serve. I'd rather have it explicitly perform actions based on input to prevent potential breaking changes down the line